src/core/container.rs

```diff
@@ -86,10 +90,9 @@ where
         &self.image
     }
 
-    fn block_until_ready(&self) {
-        log::debug!("Waiting for container {} to be ready", self.id);
-
-        for condition in self.image.ready_conditions() {
+    pub fn block_until_ready(&self, ready_conditions: Vec<WaitFor>) {
```

Comment:
I want as many of these functions as possible to remain private, otherwise any change to them is a breaking change.